kratos/CMakeLists.txt

```diff
@@ -2,188 +2,39 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/time_discretization.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/constraint_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/sensitivity_builder.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/mortar_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/exact_mortar_segmentation_utility.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/timer.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/geometrical_transformation_utilities.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/delaunator_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/properties_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/string_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/result_dabatase.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/geometry_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/normal_calculation_utils.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/tetrahedra_ball.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/tetrahedra_edge_shell.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/cad_io_modeler.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modeler/modeler_factory.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/triangle_2d_3_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/triangle_2d_3_ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/modified_shape_functions/tetrahedra_3d_4_ausas_modified_shape_functions.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/elements/embedded_nodal_variable_calculation_element_simplex.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/elements/mesh_element.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/conditions/mesh_condition.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/processes/apply_periodic_boundary_condition_process.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/linear_solvers/amgcl_solver_impl.cpp;
-    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/communication_coloring_utilities.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/constraints/slip_constraint.cpp
+SET(LIST_OF_SOURCE_EXCEPTIONS
+    ${CMAKE_CURRENT_SOURCE_DIR}/processes/assign_scalar_field_to_entities_process.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sources/kratos_version.cpp # kratos_version is excluded because I separated it to another CU so you didn't have to recompile the whole core every time the version changed, just relink it. In order to not pollute with more .dll / .so a separate .o is created and then using during the linking of KratosCore.dll
+    ${CMAKE_CURRENT_SOURCE_DIR}/processes/mesh_smoothing_process.cpp # This wasn't in the original list for some reason
+    ${CMAKE_CURRENT_SOURCE_DIR}/sources/kratos_exception.cpp # This wasn't in the original list for some reason
```

Comment:
what happens if you add it to the list? (excluding from cotire if necessary)

Reply:
We can try

Reply:
Let's see the CI

Reply:
for `kratos_exception.cpp` only minor fixes are necessary
can you also add `mesh_smoothing_process.cpp`?

Reply:
> can you also add mesh_smoothing_process.cpp?

hm I just checked, it is not included / used anywhere so I suggest to remove it completely
can you make a separate PR please?